adapters/Unity/UnityAdapter/GADMediationAdapterUnity.m

```diff
@@ -14,83 +14,125 @@
+@property (nonatomic, strong) UADSBannerView *bannerView;
 @end
 
 @implementation GADMediationAdapterUnity
 
 // Called on Admob->init
-+ (void)setUpWithConfiguration:(GADMediationServerConfiguration *)configuration
-             completionHandler:(GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  NSMutableSet *gameIDs = [[NSMutableSet alloc] init];
-  for (GADMediationCredentials *cred in configuration.credentials) {
-    NSString *gameIDFromSettings = cred.settings[kGADMAdapterUnityGameID];
-    GADMAdapterUnityMutableSetAddObject(gameIDs, gameIDFromSettings);
-  }
-
-  if (!gameIDs.count) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorInvalidServerParameters,
-        @"UnityAds mediation configurations did not contain a valid game ID.");
-    completionHandler(error);
-    return;
-  }
-
-  NSString *gameID = [gameIDs anyObject];
-  if (gameIDs.count > 1) {
-    NSLog(@"Found the following game IDs: %@. "
-          @"Please remove any game IDs you are not using from the AdMob UI.",
-          gameIDs);
-    NSLog(@"Initializing Unity Ads SDK with the game ID %@.", gameID);
-  }
-
-  [[GADMAdapterUnity alloc] initializeWithGameID:gameID withCompletionHandler:completionHandler];
++ (void)setUpWithConfiguration:(GADMediationServerConfiguration *)configuration completionHandler:(GADMediationAdapterSetUpCompletionBlock)completionHandler {
+    NSSet *gameIDs = configuration.gameIds;
+    if (!gameIDs.count) {
+        completionHandler([NSError noValidGameId]);
+        return;
+    }
+    
+    NSString *gameID = [gameIDs anyObject];
+    if (gameIDs.count > 1) {
+        NSLog(@"Found the following game IDs: %@. "
+              @"Please remove any game IDs you are not using from the AdMob UI.",
+              gameIDs);
+        NSLog(@"Initializing Unity Ads SDK with the game ID %@.", gameID);
+    }
+    
+    [[GADUnityRouter sharedRouter] sdkInitializeWithGameId:gameID withCompletionHandler:completionHandler];
 }
 
 + (GADVersionNumber)adSDKVersion {
-  GADVersionNumber version = {0};
-  NSString *sdkVersion = [UnityAds getVersion];
-  NSArray<NSString *> *components = [sdkVersion componentsSeparatedByString:@"."];
-  if (components.count == 3) {
-    version.majorVersion = components[0].integerValue;
-    version.minorVersion = components[1].integerValue;
-    version.patchVersion = components[2].integerValue;
-  } else {
-    NSLog(@"Unexpected Unity Ads version string: %@. Returning 0 for adSDKVersion.", sdkVersion);
-  }
-  return version;
+    GADVersionNumber version = {0};
+    NSString *sdkVersion = [UnityAds getVersion];
+    NSArray<NSString *> *components = [sdkVersion componentsSeparatedByString:@"."];
+    if (components.count == 3) {
+        version.majorVersion = components[0].integerValue;
+        version.minorVersion = components[1].integerValue;
+        version.patchVersion = components[2].integerValue;
+    } else {
```

Comment:
```objc
GADVersionNumber extractVersionFromString(NSString * string, Bool: addBuildNumber) {
     GADVersionNumber version = {0};
     NSArray<NSString *> *components = [string componentsSeparatedByString:@"."];
      if (components.count > 3) {
          version.majorVersion = components[0].integerValue;
          version.minorVersion = components[1].integerValue;
          NSInteger patch = components[2].integerValue
          version.patchVersion = addBuildNumber ? patch * 100 + components[3].integerValue : patch;
      } 
      return  version
}
```

and  then use it like this 

``` objc 
+ (GADVersionNumber)adSDKVersion {
   return extractVersionFromString([UnityAds getVersion], false);
}


+ (GADVersionNumber)adapterVersion {
 return extractVersionFromString(kGADMAdapterUnityVersion, true)
}
```
**P.S.** can be either a function or a helper static method. I just tried to show a quick idea